rotkehlchen/db/history_events.py

```diff
@@ -397,39 +401,55 @@ def get_history_events(
         TODO: To not query all columns with all joins for all cases, we perhaps can
         peek on the entry type of the filter and adjust the SELECT fields accordingly?
         """
+        bindings, type_idx = [], 0
         base_prefix = 'SELECT '
         base_suffix = f'{HISTORY_BASE_ENTRY_FIELDS}, {EVM_EVENT_FIELDS}, {ETH_STAKING_EVENT_FIELDS} {ALL_EVENTS_DATA_JOIN}'  # noqa: E501
-        bindings = []
-        type_idx = 1 if group_by_event_ids is True else 0
-        special_free_query = False
-        free_query_group_by = maybe_filter_ignore_asset(filter_query)
+        group_by_event_id_query = (
+            f'{base_prefix} COUNT(*), * FROM (SELECT {base_suffix}'
+            '{exclude_ignore_assets}) {filters} {limit}'
+        )
+        ignore_asset_filter = maybe_filter_ignore_asset(filter_query, negation=True)
+        if ignore_asset_filter != '':
+            ignore_asset_filter = f' WHERE event_identifier NOT IN (SELECT event_identifier FROM history_events {ignore_asset_filter})'  # noqa: E501
 
-        if has_premium is True:
-            if group_by_event_ids is True:
-                base_prefix += 'COUNT(*), '
-            base_query = f'{base_prefix} {base_suffix}'
+        if group_by_event_ids:
+            type_idx = 1
+            filters, filters_bindings = filter_query.prepare(
+                with_order=True,
+                with_group_by=True,
+                with_pagination=False,
+                without_ignored_asset_filter=True,
+            )
         else:
-            if group_by_event_ids is True:
-                special_free_query = True  # a bit ugly conditions to keep limit at groups for free users  # noqa: E501
-                free_query_group_by += 'GROUP BY event_identifier'
-                base_query = (
-                    f'{base_prefix} * FROM (SELECT COUNT(*), {base_suffix} '
-                    f'{free_query_group_by} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) '
+            filters, filters_bindings = filter_query.prepare(with_pagination=False)
+
+        if has_premium:
+            if group_by_event_ids:
+                base_query = group_by_event_id_query.format(
+                    exclude_ignore_assets=ignore_asset_filter,
```

Comment:
`exclude_ignore_assets` makes me thing it is a boolean but instead is a string filter. Can you rename it to explicit that is a filter?